build.c

```diff
@@ -199,6 +202,46 @@ buildadd(struct node *n)
 	e->flags &= ~FLAG_CYCLE;
 }
 
+static void
+statusfmt(char *fmt)
+{
+	struct timespec ctime;
+
+	for (; *fmt; ++fmt)
+		if (*fmt == '%')
+			switch (*(++fmt)) {
+			case 's':
+				printf("%zu", nstarted);
+				break;
+			case 'f':
+				printf("%zu", nfinished);
+				break;
+			case 't':
+				printf("%zu", ntotal);
+				break;
+			case 'r':
+				printf("%zu", nstarted - nfinished);
+				break;
+			case 'u':
+				printf("%zu", ntotal - nstarted);
+				break;
+			case 'p':
+				printf("%3lu%%", 100 * nfinished / ntotal);
+				break;
+			case 'e':
+				clock_gettime(CLOCK_REALTIME, &ctime);
+				printf("%.3f",
+					((double)ctime.tv_sec + (double)ctime.tv_nsec * 0.000000001) -
```

Comment:
I don't think these casts are necessary. The `0.000000001` literal has type `double`, and usual arithmetic conversions will convert the integers to floating point for the binary operators.